internal/extproc/watcher.go

```diff
@@ -66,9 +68,54 @@ func (cw *configWatcher) loadConfig() error {
 	}
 	cw.lastMod = stat.ModTime()
 	cw.l.Info("loading a new config", slog.String("path", cw.path))
+
+	// Re-hydrate the current config file for later diffing.
+	previous := cw.current
+	current, err := cw.getConfig()
+	if err != nil {
+		return fmt.Errorf("failed to read the config file: %w", err)
+	}
+
+	// Print the diff between the old and new config.
+	cw.diff(previous, current)
```

Comment:
can you avoid calling this function when it's not debugging mode by using  `cw.l.Enabled(ctx, slog.LevelDebug)`